src/relax/ir/binding_rewrite.cc

```diff
@@ -212,17 +213,21 @@ class RemoveUnusedVars : public ExprMutator {
               if (kv.second.empty() &&  // kv.first is not used by fn outputs.
                   fn_outputs.end() == std::find(fn_outputs.begin(), fn_outputs.end(), kv.first)) {
                 unused.push_back(kv.first);
+              } else {
+                users_keys.push_back(kv.first);
               }
             }
 
             for (size_t i = prev_size; i < unused.size(); ++i) {
               users.erase(unused[i]);
               // remove def site.
-              for (auto kv : users) {  // remove use site.
-                auto it = std::find(kv.second.begin(), kv.second.end(), unused[i]);
-                if (it != kv.second.end()) {
-                  kv.second.erase(it);
-                  users.Set(kv.first, std::move(kv.second));
+              for (const auto& key: users_keys) {  // remove use site.
+                ICHECK(users.count(key)) << "remove unused algorithm internal error.";
```

Comment:
This prompt is too general. For this case where the intention of ICHECK is very obvious, we actually either prompt nothing or prompt some more detailed information, like “the the key is expected to be in the mapping `users`”.